### ISSUE TITLE
`prevSibling` can return an invalid BufferNode

### ISSUE THREAD

**Issue by mhuebert** [lezer-parser/lezer#36]
While migrating to the latest ClojureScript/Lezer versions, I'm finding some odd `BufferNode` instances returned by `prevSibling`, where `type` is undefined and `index` is `-1`.

The following example is in Clojure but should be straightforward. I parse the text `1 a`, navigate to the `a` Symbol node, and try navigating leftward via `prevSibling`. I would expect to find the Number node for `1`. (the particular types aren't relevant here, I just picked two different type so I can easily know which thing I'm looking at):

```clj 
(let [state (test-utils/make-state #js[(syntax lezer-clj/parser)] "1 a")]
  (-> (.-tree state)
      (.resolve 2 1) ;; Symbol "a"
      .-prevSibling))
```

Logging the result I see:

```js 
BufferNode {context: {…}, _parent: null, index: -1, type: undefined}
  context: {parent: TreeNode, buffer: TreeBuffer, index: 0, start: 0}
  cursor: TreeCursor
  firstChild: BufferNode
  from: 26
  index: -1
  lastChild: (...)
  name: (...)
  nextSibling: (...)
  parent: (...)
  prevSibling: (...)
  to: (...)
  type: undefined
  _parent: null
  __proto__: Object
```

Prior to the `nextSibling` interface I was using something roughly like:

```
function left(node) { 
  return node.parent.childBefore(node.from) 
}
```

**Comment by marijnh** [comment]
[This patch](https://github.com/lezer-parser/lezer-tree/commit/cdb5d29cd20be32f45d60a2f0f1755da52d28f13) should help with this. Was a dumb bug where I didn't update one call site when I had changed an internal interface. I've released lezer-tree 0.12.2 with the fix.

**Comment by mhuebert** [comment]
Thanks!
